### PR TITLE
Adding wallpaperdownloader snap package configuration

### PR DESCRIPTION
**Pull request by egara**
Hi!

I would like to add WallpaperDownloader Java (Swing) application to the repository. The directory (wallpaperdownloader) contains all the stuff needed to build a snap package for this application (JRE 8) which has a GUI and it is using Java Swing.

Thanks!

Best,

Eloy

---
This change is [Reviewable](https://reviewable.io/reviews/ubuntu/snappy-playpen/127)

### CODE REVIEW

File: wallpaperdownloader/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: wallpaperdownloader
+version: 2.0
```
Comment (tsimonq2) [inline wallpaperdownloader/snapcraft.yaml]: Could you add some double quotes around the version?

File: wallpaperdownloader/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+summary: Download and manage your favorite wallpapers from the Internet
+description:
```
Comment (tsimonq2) [inline wallpaperdownloader/snapcraft.yaml]: `description: |`

File: wallpaperdownloader/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  wallpaperdownloader:
+    command: wallpaperdownloader.sh
+    plugs: [x11, network-bind, home]
+
+parts:
+  # Pulls the code from the original source (master branch)
+  wallpaperdownloader:
+    plugin: maven
+    source: https://bitbucket.org/eloy_garcia_pca/wallpaperdownloader.git
+    source-type: git
```
Comment (tsimonq2) [inline wallpaperdownloader/snapcraft.yaml]: You don't need this line because `source:` ends with `.git`

File: wallpaperdownloader/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# WallpaperDownloader snap
+
+This project creates a working snap of WallpaperDownloader application.
```
Comment (tsimonq2) [inline wallpaperdownloader/README.md]: Something grammatical ;)

`This project creates a working snap of the WallpaperDownloader application.`

File: wallpaperdownloader/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+WallpaperDownloader is a simple GUI Java based application to download wallpapers from the Internet and manage them.
```
Comment (tsimonq2) [inline wallpaperdownloader/README.md]: `Java-based`

File: wallpaperdownloader/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# WallpaperDownloader snap
+
+This project creates a working snap of WallpaperDownloader application.
+WallpaperDownloader is a simple GUI Java based application to download wallpapers from the Internet and manage them. 
+For more information, please visit the official [Wiki](https://bitbucket.org/eloy_garcia_pca/wallpaperdownloader/wiki/Home) of the project.
+
+## Current state
+
+Working features:
+ - The applications is working fine. You can even change the downloads folder (by default inside snap directory) to a directory within the home's user.
```
Comment (tsimonq2) [inline wallpaperdownloader/README.md]: `application`

`user's home folder` - maybe? I can't tell what you are trying to get at here

File: wallpaperdownloader/README.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+Known issues:
+  - Open downloads folder button is not working. An exception is thrown:
```
Comment (tsimonq2) [inline wallpaperdownloader/README.md]: ``Open downloads folder``

**Comment by tsimonq2** [comment]
Nice work @egara!

I added some comments that ensure it uses the same format as the other snaps in the Playpen.

Thanks again! :)

**Comment by egara** [comment]
Thank you very much for the corrections :) 

Do I have to create another new Pull Request with all these modifications included?

**Comment by tsimonq2** [comment]
@egara no you don't, the commits you pushed are updated here, someone with access can merge as soon as they confirm that it is good to go :)

File: wallpaperdownloader/README.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+WallpaperDownloader is a simple GUI Java-based application to download wallpapers from the Internet and manage them. 
+For more information, please visit the official [Wiki](https://bitbucket.org/eloy_garcia_pca/wallpaperdownloader/wiki/Home) of the project.
+
+## Current state
+
+Working features:
+ - The application is working fine. You can even change the downloads folder (by default inside snap directory) to a directory within the user's home folder.
+
+Known issues:
+  - Open 'downloads folder' button is not working. An exception is thrown:
```
Comment (tsimonq2) [inline wallpaperdownloader/README.md]: @egara not a ' , a ` will do the highlighting in GitHub

**Comment by egara** [comment]
Thank you very much again :)

File: wallpaperdownloader/README.md (hunk 8 above)
Comment (egara) [inline wallpaperdownloader/README.md]: Mmmm ok! Now I get it. I'll modify it right now

**Comment by tsimonq2** [comment]
@egara I'll test locally before I give a thumbs up, thanks again for your contribution! :)

**Comment by tsimonq2** [comment]
:+1: LGTM

**Comment by tsimonq2** [comment]
Reviewed 3 of 5 files at r1, 1 of 1 files at r2, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127)*

**Comment by egara** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[wallpaperdownloader/README.md, line 3 \[r1\]](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127#-KLQl4f8EpTzUHvnoDRG-r1-3:-KLQxXEW_AjOksoCkxzI:-499146009) ([raw file](https://github.com/ubuntu/snappy-playpen/blob/74a2de64fa70cd2b5c1baded14cd4b98be8279a3/wallpaperdownloader/README.md#L3)):*
<details><summary><i>Previously, tsimonq2 (Simon Quigley) wrote…</i></summary>
> Something grammatical ;)
> 
> 
> 
> `This project creates a working snap of the WallpaperDownloader application.`
</details>
Done.

---

*[wallpaperdownloader/README.md, line 4 \[r1\]](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127#-KLQl4f8EpTzUHvnoDRG-r1-4:-KLQxZ2Jkf_pswtry4Ho:-499146009) ([raw file](https://github.com/ubuntu/snappy-playpen/blob/74a2de64fa70cd2b5c1baded14cd4b98be8279a3/wallpaperdownloader/README.md#L4)):*
<details><summary><i>Previously, tsimonq2 (Simon Quigley) wrote…</i></summary>
> `Java-based`
</details>
Done.

---

*[wallpaperdownloader/README.md, line 10 \[r1\]](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127#-KLQl4f8EpTzUHvnoDRG-r1-10:-KLQxgxc2u5DBLpBUyJ8:-499146009) ([raw file](https://github.com/ubuntu/snappy-playpen/blob/74a2de64fa70cd2b5c1baded14cd4b98be8279a3/wallpaperdownloader/README.md#L10)):*
<details><summary><i>Previously, tsimonq2 (Simon Quigley) wrote…</i></summary>
> `application`
> 
> 
> 
> `user's home folder` - maybe? I can't tell what you are trying to get at here
</details>
Done.

---

*[wallpaperdownloader/README.md, line 13 \[r1\]](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127#-KLQl4f8EpTzUHvnoDRG-r1-13:-KLQxiAnxGfn8lfYueZn:-499146009) ([raw file](https://github.com/ubuntu/snappy-playpen/blob/74a2de64fa70cd2b5c1baded14cd4b98be8279a3/wallpaperdownloader/README.md#L13)):*
<details><summary><i>Previously, tsimonq2 (Simon Quigley) wrote…</i></summary>
> ``Open downloads folder`` - with ` around it
</details>
Done.

---

*[wallpaperdownloader/README.md, line 13 \[r3\]](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127#-KLQl4f8EpTzUHvnoDRG-r3-13:-KLQxjtaNhdgn4HsTSv4:-499146009) ([raw file](https://github.com/ubuntu/snappy-playpen/blob/810dc29f079b8c4a1afaeca56a3f8ff515a5d3c0/wallpaperdownloader/README.md#L13)):*
<details><summary><i>Previously, egara (Eloy García Almadén) wrote…</i></summary>
> Mmmm ok! Now I get it. I'll modify it right now
</details>
Done.

---

*[wallpaperdownloader/snapcraft.yaml, line 2 \[r1\]](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127#-KLQl4f8EpTzUHvnoDRI-r1-2:-KLQxliR6eQO4wr_KmJ2:-499146009) ([raw file](https://github.com/ubuntu/snappy-playpen/blob/74a2de64fa70cd2b5c1baded14cd4b98be8279a3/wallpaperdownloader/snapcraft.yaml#L2)):*
<details><summary><i>Previously, tsimonq2 (Simon Quigley) wrote…</i></summary>
> Could you add some double quotes around the version?
</details>
Done.

---

*[wallpaperdownloader/snapcraft.yaml, line 4 \[r1\]](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127#-KLQl4f8EpTzUHvnoDRI-r1-4:-KLQxoACaHG_qXe1cJ7-:-499146009) ([raw file](https://github.com/ubuntu/snappy-playpen/blob/74a2de64fa70cd2b5c1baded14cd4b98be8279a3/wallpaperdownloader/snapcraft.yaml#L4)):*
<details><summary><i>Previously, tsimonq2 (Simon Quigley) wrote…</i></summary>
> `description: |`
</details>
Done.

---

*[wallpaperdownloader/snapcraft.yaml, line 18 \[r1\]](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127#-KLQl4f8EpTzUHvnoDRI-r1-18:-KLQxq2Be7EFqUUxj5fT:-499146009) ([raw file](https://github.com/ubuntu/snappy-playpen/blob/74a2de64fa70cd2b5c1baded14cd4b98be8279a3/wallpaperdownloader/snapcraft.yaml#L18)):*
<details><summary><i>Previously, tsimonq2 (Simon Quigley) wrote…</i></summary>
> You don't need this line because `source:` ends with `.git`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127)*

**Comment by dholbach** [comment]
:lgtm:

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/127#-:-KLRQPzm8VQv4V4q6t62:1417088721)*

**Pull request merged**